### PR TITLE
Test suite for `safeTransferFrom`

### PR DESCRIPTION
**Pull request by saturnial**

### CODE REVIEW

File: test/ts/test_utils/send_transactions.ts [hunk 1]
```diff
@@ -26,8 +26,8 @@ function findMethod(abi: Web3.AbiDefinition[], name: string, inputTypes: string)
 // method of a contract.
 // Truffle contract instance cannot handle overloaded methods
 // (truffle will only handle the first implementation of the method).
-export default function sendTransaction(
-    truffleContractInstance: any,
+export function sendTransaction(
+    truffleContractInstance: Web3.ContractInstance,
```
Comment (nadavhollander) [inline test/ts/test_utils/send_transactions.ts]: Is this supposed to be a truffle contract instance?  Or a web3 contract instance?  Those are two very different things -- we should make sure the key and value are consistent.

File: test/ts/unit/debt_token.ts [hunk 2]
```diff
@@ -29,6 +31,9 @@ const repaymentRouterContract = artifacts.require("RepaymentRouter");
 contract("Debt Token (Unit Tests)", (ACCOUNTS) => {
     let debtToken: DebtTokenContract;
 
+    let debtTokenTruffle: Web3.ContractInstance;
```
Comment (nadavhollander) [inline test/ts/unit/debt_token.ts]: Ditto on the above.

File: test/ts/unit/debt_token.ts [hunk 3]
```diff
@@ -1429,4 +1437,125 @@ contract("Debt Token (Unit Tests)", (ACCOUNTS) => {
...
+                            TOKEN_OWNER_1,
+                            toReceiver ? receiver.address : mockRegistry.address,
+                            tokenID,
+                        ),
+                    ).to.eventually.be.rejectedWith(REVERT_ERROR);
+                });
+            });
+        };
+
+        describe("to a valid receiver contract", () => {
```
Comment (nadavhollander) [inline test/ts/unit/debt_token.ts]: Please include tests for:

1. Transferring to non-contract accounts
2. Transferring under approval conditions -- i.e. ACCOUNT_1 approved ACCOUNT_2 for X tokens, ACCOUNT_2 calls safeTransferFrom(ACCOUNT_1, Y);

**Comment by nadavhollander** [comment]
I would also emphasize that, given that we're using a fairly well tested implementation anyways, we don't have to get _too_ thorough in this test suite, and I would up-prioritize the corresponding `dharma.js` PT task [here]( https://www.pivotaltracker.com/story/show/155375378)

**Comment by nadavhollander** [comment]
@saturnial Can you please make the above fixes (except for the adding of additional tests) and make tasks for the remaining tests post dharma.js@0.1.0?

Blocked on getting this merged in because I need the mock ERC721 receiver

File: test/ts/test_utils/send_transactions.ts (hunk 1 above)
Comment (saturnial) [reply test/ts/test_utils/send_transactions.ts]: @t4sk labeled this param as a truffle contract instance. To my understanding, it's a Web3 contract instance.

File: test/ts/unit/debt_token.ts (hunk 3 above)
Comment (saturnial) [reply test/ts/unit/debt_token.ts]: Pivotal story here: https://www.pivotaltracker.com/story/show/156604273

**Comment by saturnial** [comment]
:recycle:

**Comment by nadavhollander** [comment]
LGTM! Thanks for getting this together. Please merge in at your convenience :)

**Pull request merged**